### PR TITLE
[PSR-15] Adds question/answer to section 6.2 Middleware Design

### PR DESCRIPTION
**Pull request by weierophinney**
Details the relationship between middleware and handlers, per feedback received on the mailing list. Specfically, it details the three different potential general paths for middleware, and request handler designs that might facilitate those.

### CODE REVIEW

File: proposed/http-handlers/request-handlers-meta.md [hunk 1]
```diff
@@ -318,6 +318,51 @@ to define a standard that is specific to the nature of asynchronous middleware.
 _See "client vs server side middleware" in [relevant links](#8-relevant-links) for
 additional information._
 
+#### What is the role of the request handler?
+
+Middleware has the following roles:
+
+- Producing a response on its own. If specific request conditions are met, the
+  middleware an produce and return a response.
```
Comment (oscarotero) [inline proposed/http-handlers/request-handlers-meta.md]: I guess you mean "middleware can produce" :)

File: proposed/http-handlers/request-handlers-meta.md [hunk 2]
```diff
@@ -318,6 +318,51 @@ to define a standard that is specific to the nature of asynchronous middleware.
...
+// Capturing the response to manipulate:
+$response = $handler->handle($request);
+```
+
+How the handler acts is entirely up to the developer, so long as it produces a
+response.
+
+In one common scenario, the handler implements a _queue_ or a _stack_ of
+middleware instances internally. In such cases, calling
+`$handler->handle($request)` will pull the advance the internal pointer, pull
```
Comment (stefanotorresi) [inline proposed/http-handlers/request-handlers-meta.md]: it think this needs rephrasing.

File: proposed/http-handlers/request-handlers-meta.md (hunk 1 above)
Comment (weierophinney) [reply proposed/http-handlers/request-handlers-meta.md]: Fixed!

File: proposed/http-handlers/request-handlers-meta.md (hunk 2 above)
Comment (weierophinney) [reply proposed/http-handlers/request-handlers-meta.md]: Fixed!

**Pull request merged**